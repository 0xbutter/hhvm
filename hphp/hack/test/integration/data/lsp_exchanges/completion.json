[
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "initialize",
  "params":
    {
      "initializationOptions":{},
      "processId":null,
      "rootPath":"${root_path}",
      "capabilities":{}
    }
},
{
  "jsonrpc": "2.0",
  "method": "textDocument/didOpen",
  "params": {
     "textDocument": {
       "uri":"${php_file_uri}",
       "languageId": "hack",
       "version":1,
       "text":"${php_file}"
     }
   }
},
{
  "comment": "Add '$x = <'",
  "jsonrpc": "2.0",
  "method": "textDocument/didChange",
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "contentChanges": [{
         "range": {
           "start": {"line":3, "character":0},
           "end": {"line":3, "character":0}
         },
         "text":"$x = <"
       }]
   }
},
{
  "comment": "autocomplete after '$x = <'",
  "jsonrpc": "2.0",
  "method": "textDocument/completion",
  "id": 2,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 3, "character": 6}
   }
},
{
  "comment": "Add '$x = <a'",
  "jsonrpc": "2.0",
  "method": "textDocument/didChange",
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "contentChanges": [{
         "range": {
           "start": {"line":3, "character":0},
           "end": {"line":3, "character":6}
         },
         "text":"$x = <a"
       }]
   }
},
{
  "comment": "autocomplete after '$x = <a'",
  "jsonrpc": "2.0",
  "method": "textDocument/completion",
  "id": 3,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 3, "character": 7}
   }
},
{
  "comment": "Add '$x = <ab:cd:text '",
  "jsonrpc": "2.0",
  "method": "textDocument/didChange",
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "contentChanges": [{
         "range": {
           "start": {"line":3, "character":0},
           "end": {"line":3, "character":7}
         },
         "text":"$x = <ab:cd:text "
       }]
   }
},
{
  "comment": "autocomplete after '$x = <ab:cd:text '",
  "jsonrpc": "2.0",
  "method": "textDocument/completion",
  "id": 4,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 3, "character": 17}
   }
},
{
  "comment": "Add '$x = <ab:cd:text w'",
  "jsonrpc": "2.0",
  "method": "textDocument/didChange",
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "contentChanges": [{
         "range": {
           "start": {"line":3, "character":0},
           "end": {"line":3, "character":17}
         },
         "text":"$x = <ab:cd:text w"
       }]
   }
},
{
  "comment": "autocomplete after '$x = <ab:cd:text w'",
  "jsonrpc": "2.0",
  "method": "textDocument/completion",
  "id": 5,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 3, "character": 18}
   }
},
{
  "comment": "Add '$x = new :'",
  "jsonrpc": "2.0",
  "method": "textDocument/didChange",
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "contentChanges": [{
         "range": {
           "start": {"line":3, "character":0},
           "end": {"line":3, "character":18}
         },
         "text":"$x = new :"
       }]
   }
},
{
  "comment": "autocomplete after '$x = new :'",
  "jsonrpc": "2.0",
  "method": "textDocument/completion",
  "id": 6,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 3, "character": 10}
   }
},
{
  "comment": "Add '$x = new :a'",
  "jsonrpc": "2.0",
  "method": "textDocument/didChange",
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "contentChanges": [{
         "range": {
           "start": {"line":3, "character":0},
           "end": {"line":3, "character":10}
         },
         "text":"$x = new :a"
       }]
   }
},
{
  "comment": "autocomplete after '$x = new :a'",
  "jsonrpc": "2.0",
  "method": "textDocument/completion",
  "id": 7,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 3, "character": 11}
   }
},
{
  "comment": "Add '$x = <ab:cd:text/>; $y = $x->'",
  "jsonrpc": "2.0",
  "method": "textDocument/didChange",
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "contentChanges": [{
         "range": {
           "start": {"line":3, "character":0},
           "end": {"line":3, "character":11}
         },
         "text":"$x = <ab:cd:text/>; $y = $x->"
       }]
   }
},
{
  "comment": "autocomplete after '$x = <ab:cd:text/>; $y = $x->'",
  "jsonrpc": "2.0",
  "method": "textDocument/completion",
  "id": 8,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 3, "character": 29}
   }
},
{
  "comment": "Add '$x = <ab:cd:text/>; $y = $x->:'",
  "jsonrpc": "2.0",
  "method": "textDocument/didChange",
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "contentChanges": [{
         "range": {
           "start": {"line":3, "character":0},
           "end": {"line":3, "character":29}
         },
         "text":"$x = <ab:cd:text/>; $y = $x->:"
       }]
   }
},
{
  "comment": "autocomplete after '$x = <ab:cd:text/>; $y = $x->:'",
  "jsonrpc": "2.0",
  "method": "textDocument/completion",
  "id": 9,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "position": {"line": 3, "character": 30}
   }
},
{
  "jsonrpc": "2.0",
  "id": 99,
  "method": "shutdown",
  "params": {}
}
]
