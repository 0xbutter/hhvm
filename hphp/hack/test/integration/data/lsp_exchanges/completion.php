<?hh  //strict

function xhp_consumption(): void {

}
