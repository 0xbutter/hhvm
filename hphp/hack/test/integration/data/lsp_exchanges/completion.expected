[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": false,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<"
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "renameProvider": false,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "telemetry/event",
    "params": {
      "type": 4,
      "message": "Version in hhconfig=[NoVersion]"
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 2,
    "result": {
      "isIncomplete": false,
      "items": [
        {
          "label": "ab:cd:alpha",
          "kind": 7,
          "detail": "class",
          "inlineDetail": "class",
          "insertText": "ab:cd:alpha",
          "insertTextFormat": 1
        },
        {
          "label": "ab:cd:text",
          "kind": 7,
          "detail": "class",
          "inlineDetail": "class",
          "insertText": "ab:cd:text",
          "insertTextFormat": 1
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 3,
    "result": {
      "isIncomplete": false,
      "items": [
        {
          "label": "ab:cd:alpha",
          "kind": 7,
          "detail": "class",
          "inlineDetail": "class",
          "insertText": "ab:cd:alpha",
          "insertTextFormat": 1
        },
        {
          "label": "ab:cd:text",
          "kind": 7,
          "detail": "class",
          "inlineDetail": "class",
          "insertText": "ab:cd:text",
          "insertTextFormat": 1
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 4,
    "result": {
      "isIncomplete": false,
      "items": [
        {
          "label": "width",
          "kind": 10,
          "detail": "?int",
          "inlineDetail": "?int",
          "insertText": "width",
          "insertTextFormat": 1
        },
        {
          "label": "color",
          "kind": 10,
          "detail": "?string",
          "inlineDetail": "?string",
          "insertText": "color",
          "insertTextFormat": 1
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 5,
    "result": {
      "isIncomplete": false,
      "items": [
        {
          "label": "width",
          "kind": 10,
          "detail": "?int",
          "inlineDetail": "?int",
          "insertText": "width",
          "insertTextFormat": 1
        },
        {
          "label": "color",
          "kind": 10,
          "detail": "?string",
          "inlineDetail": "?string",
          "insertText": "color",
          "insertTextFormat": 1
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 6,
    "result": {
      "isIncomplete": false,
      "items": [
        {
          "label": ":ab:cd:alpha",
          "kind": 4,
          "detail": "function(): :ab:cd:alpha",
          "inlineDetail": "()",
          "itemType": ":ab:cd:alpha",
          "insertText": ":ab:cd:alpha",
          "insertTextFormat": 1
        },
        {
          "label": ":ab:cd:text",
          "kind": 4,
          "detail": "function(): :ab:cd:text",
          "inlineDetail": "()",
          "itemType": ":ab:cd:text",
          "insertText": ":ab:cd:text",
          "insertTextFormat": 1
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 7,
    "result": {
      "isIncomplete": false,
      "items": [
        {
          "label": ":ab:cd:alpha",
          "kind": 4,
          "detail": "function(): :ab:cd:alpha",
          "inlineDetail": "()",
          "itemType": ":ab:cd:alpha",
          "insertText": ":ab:cd:alpha",
          "insertTextFormat": 1
        },
        {
          "label": ":ab:cd:text",
          "kind": 4,
          "detail": "function(): :ab:cd:text",
          "inlineDetail": "()",
          "itemType": ":ab:cd:text",
          "insertText": ":ab:cd:text",
          "insertTextFormat": 1
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 8,
    "result": {
      "isIncomplete": false,
      "items": [
        {
          "label": ":width",
          "kind": 10,
          "detail": "?int",
          "inlineDetail": "?int",
          "insertText": ":width",
          "insertTextFormat": 1
        },
        {
          "label": ":color",
          "kind": 10,
          "detail": "?string",
          "inlineDetail": "?string",
          "insertText": ":color",
          "insertTextFormat": 1
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 9,
    "result": {
      "isIncomplete": false,
      "items": [
        {
          "label": ":width",
          "kind": 10,
          "detail": "?int",
          "inlineDetail": "?int",
          "insertText": ":width",
          "insertTextFormat": 1
        },
        {
          "label": ":color",
          "kind": 10,
          "detail": "?string",
          "inlineDetail": "?string",
          "insertText": ":color",
          "insertTextFormat": 1
        }
      ]
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 99,
    "result": null
  }
]
